script.transmission/resources/lib/player.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2013 Paul Price, Artem Glebov
+
+import os
+import sys
+from kodi_six import xbmc, xbmcaddon, xbmcgui
+from six import iteritems
+import transmissionrpc
+
+__settings__ = xbmcaddon.Addon(id='script.transmission')
+
+BASE_RESOURCE_PATH = xbmc.translatePath( os.path.join( __settings__.getAddonInfo('path'), 'resources', 'lib' ) )
+sys.path.append (BASE_RESOURCE_PATH)
+
+import common
+
+class SubstitutePlayer(xbmc.Player):
+    def __init__(self):
+        xbmc.Player.__init__(self)
+        self.prev_settings = {}
+        self.refreshSettings()
+
+    def onPlayBackStarted(self):
+        self.refreshSettings()
+        if self.active and xbmc.Player().isPlayingVideo():
```

Comment:
It is not guaranteed that when this event is called (player initialization) player is already playingvideo, another reason to change it to onAVStarted